kernelci/cli/user.py

```diff
@@ -115,127 +110,94 @@ def add(username, email, config,  # pylint: disable=too-many-arguments
         'username': username,
         'email': email,
         'password': password,
-        'groups': group
     }
     configs = kernelci.config.load(config)
     api_config = configs['api'][api]
     api = kernelci.api.get_api(api_config, secrets.api.token)
-    data = api.create_user(user)
-    click.echo(json.dumps(data, indent=indent))
+    api.create_user(user)
 
 
 @kci_user.command
-@click.argument('email')
```

Comment:
We can send `email` directly for the verification. Why send username instead?

Reply:
Because it's easier from a user point of view to provide the username than the email, if the main way of referring to a user is their username.

I think an improvement here might be to have something that can be either a username or an email e.g. a `user` argument and then try to find a user with either the username or the email matching the provided string.  A username doesn't allow characters such as `@` (if I remember correctly) so we can't have a username colliding with another user's email - or we may want to have a unique index on these 2 fields anyway.

Reply:
The key thing here is that the CLI needs to be optimal from a user point of view, and it shouldn't be derived from the API endpoints or bindings or how things might be implemented.  Ideally these things would line up neatly, but whenever there's a gap then usability on the command line should be the priority.  So if a user is more likely to know their username than their email address, or typically would have a shorter username than email address, then relying on the username everywhere is better even if it means an extra query to the API (and this isn't meant to be a high-traffic endpoint like the web dashboard would be doing).

Reply:
IMO username and email both are key fields of any user account. Hence providing email directly would be as easy as providing a username.  Anyway if providing username seems optimal from user's pov, then it's fine.

Reply:
> I think an improvement here might be to have something that can be either a username or an email e.g. a user argument and then try to find a user with either the username or the email matching the provided string. A username doesn't allow characters such as @ (if I remember correctly) so we can't have a username colliding with another user's email - or we may want to have a unique index on these 2 fields anyway.

The provision of letting the user provide username or email would complicate the code. I think it should allow just one specific argument.

Reply:
Well it's not that hard:
```python
    users = (
        api.user.find({"username": username}) or
        api.user.find({'email': email})
    )
```